api/go.mod

```diff
@@ -4,18 +4,18 @@ go 1.18
 
 require (
 	github.com/go-logr/logr v1.2.4
-	github.com/metal3-io/baremetal-operator/apis v0.3.1
+	github.com/metal3-io/baremetal-operator/apis v0.4.0
 	github.com/onsi/ginkgo/v2 v2.12.0
 	github.com/onsi/gomega v1.27.10
 	github.com/openstack-k8s-operators/lib-common/modules/common v0.1.0
-	k8s.io/api v0.26.8
-	k8s.io/apimachinery v0.26.8
-	sigs.k8s.io/controller-runtime v0.14.6
+	k8s.io/api v0.27.2
```

Comment:
the k8s.io bump needs active adaptation work first. So we cannot bump to metal3 0.4. without manually bumping to k8s.io to 0.27.